framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -413,21 +414,83 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
 
         // Check type-system specific annotations.
         Class<PostconditionAnnotation> metaAnnotation = PostconditionAnnotation.class;
+        Class<String> stringType = String.class;
+        Class<AnnotationMirror> annotationmirrorType = AnnotationMirror.class;
         List<Pair<AnnotationMirror, AnnotationMirror>> declAnnotations =
                 factory.getDeclAnnotationWithMetaAnnotation(methodElement, metaAnnotation);
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
+
+            result.addAll(
+                    getPostConditionAnnotations(anno, metaAnno, annotationmirrorType, stringType));
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns postcondition annotations.
+     *
+     * @param anno the type qualifier that {@code methodElement} should have
+     * @param metaAnno postCondition meta-annotation
+     * @param annotationmirrorType the annotationMirror class type used to cast the element value
+     * @param stringType the string class type used to cast the element value
+     */
+    private Set<Postcondition> getPostConditionAnnotations(
+            AnnotationMirror anno,
+            AnnotationMirror metaAnno,
+            Class<AnnotationMirror> annotationmirrorType,
+            Class<String> stringType) {
+
+        @SuppressWarnings("unchecked")
+        List<AnnotationValue> la =
+                AnnotationUtils.getElementValue(anno, "value", List.class, false);
+
+        Set<Postcondition> result = new LinkedHashSet<>();
+
+        // Check for multiple contracts.
+        List<AnnotationMirror> annotations = new ArrayList<>(la.size());
+        for (AnnotationValue a : la) {
+            if (a.getClass().getName().equals("javax.lang.model.element.AnnotationMirror")
+                    || a.getClass()
+                            .getName()
+                            .equals("com.sun.tools.javac.code.Attribute$Compound")) {
+                annotations.add(annotationmirrorType.cast(a.getValue()));
+            }
+        }
+
+        for (AnnotationMirror a : annotations) {
+            List<String> expression =
+                    AnnotationUtils.getElementValueArray(a, "value", String.class, false);
+            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, a);
             if (postcondAnno == null) {
                 continue;
             }
+            for (String expr : expression) {
+                result.add(new Postcondition(expr, postcondAnno, a));
+            }
+        }
+
+        // Check for a single contract.
```

Comment:
Can you explain what is happening here? `la` didn't get re-assigned, so why another iteration over it?
Couldn't you merge this loop into the other loop over `la` and cover the two possible cases once?

Reply:
> Can you explain what is happening here? 

It is adding all the annotation value to its list if a single annotation is present.
> `la` didn't get re-assigned, so why another iteration over it?
 Couldn't you merge this loop into the other loop over `la` and cover the two possible cases once?

Yes, you are right. I have merged both the cases in a single loop.